### PR TITLE
V3 code review

### PR DESCRIPTION
**Pull request by patkkka**
For code reviewers:
We added:

- server class which accepts connections from clients and creates new thread for each client
- tests for those functionalities

### CODE REVIEW

File: src/test/java/com/spanish_inquisition/battleship/MainTest.java [hunk 1]
```diff
@@ -1,4 +0,0 @@
-package com.spanish_inquisition.battleship;
-
-public class MainTest {
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/MainTest.java]: 👍

File: src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+public class BattleshipServerTest {
+    private ServerSocket serverSocket;
+    private final String SERVER_ADDRESS = "localhost";
+    private final int TEST_PORT = 6660;
+    private Socket client1;
+    private Socket client2;
+
+    @BeforeClass
+    public void beforeClass() {
+        initializeLogger();
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: Logger in tests? Why? All output is in the console anyway when something goes wrong and when everything works well then why bother

File: src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        // Then
+        Assert.assertEquals(2, testClients.size());
+    }
+
+    @Test
+    public void shouldCreateServerSocket() {
+
+        // When
+        ServerSocket testServerSocket = BattleshipServer.createServerSocket();
+
```
Comment (mprtcz) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: Reduce those spaces between lines, it's hideous

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        ServerSocket serverSocket = createServerSocket();
+        clients = connectWithPlayers(serverSocket);
+        connectWithPlayers(serverSocket);
+    }
+
+    static List<Socket> connectWithPlayers(ServerSocket serverSocket) {
+        List<Socket> clientSockets = new ArrayList<>();
+        for (int i = 0; i < NUMBER_OF_PLAYERS; i++) {
+            new Thread(()->{
+                System.out.println("Hello from thread " + Thread.currentThread().getName());
```
Comment (kukulastudent) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java]: try use loger

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import static com.spanish_inquisition.battleship.common.AppLogger.initializeLogger;
+import static com.spanish_inquisition.battleship.common.AppLogger.logger;
+
+public class BattleshipServer {
+    private static final int NUMBER_OF_PLAYERS = 2;
+    private static final Integer PORT_NUMBER = 6666;
+    private static List<Socket> clients;
+
+
+    public static void main(String[] args) throws IOException {
```
Comment (kukulastudent) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java]: put psvm to other class

File: src/test/java/com/spanish_inquisition/battleship/common/AppLoggerTest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+/**
+ * @author Michal_Partacz
+ */
+public class AppLoggerTest {
+
+    @Test
+    public void testInitializeLogger() throws Exception {
```
Comment (kukulastudent) [inline src/test/java/com/spanish_inquisition/battleship/common/AppLoggerTest.java]: it's good to write what is expected, in method name or with //given //when //then

File: src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        // Then
+        Assert.assertNotNull(testServerSocket);
+
```
Comment (kukulastudent) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: :+1:  given when then

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+import static com.spanish_inquisition.battleship.common.AppLogger.defaultLevel;
+import static com.spanish_inquisition.battleship.common.AppLogger.initializeLogger;
+import static com.spanish_inquisition.battleship.common.AppLogger.logger;
+
+public class BattleshipServer {
+    private static final int NUMBER_OF_PLAYERS = 2;
+    private static final Integer PORT_NUMBER = 6666;
+    private static List<Socket> clients;
+
+
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java]: remove one line

File: src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.List;
+
+import static com.spanish_inquisition.battleship.common.AppLogger.defaultLevel;
+import static com.spanish_inquisition.battleship.common.AppLogger.initializeLogger;
+import static com.spanish_inquisition.battleship.common.AppLogger.logger;
+
+public class BattleshipServerTest {
```
Comment (kukulastudent) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: I think that it's integration test. Name of the class could have "IT" at end of name.

Comment (kukulastudent) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: http://maven.apache.org/surefire/maven-failsafe-plugin/examples/inclusion-exclusion.html

Comment (kukulastudent) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: it's only example

File: src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    public void afterMethod() {
+        try {
+            serverSocket.close();
+            client1.close();
+            client2.close();
+        } catch (IOException e) {
+            logger.log(defaultLevel, "Sockets didn't close.");
+        }
+    }
+}
```
Comment (kukulastudent) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: :+1: for check connection

File: src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+                try {
+                    client2 = new Socket(SERVER_ADDRESS, TEST_PORT);
+                } catch (IOException e) {
+                    logger.log(defaultLevel, "shouldConnectWithTwoPlayers -> cannot run client2 thread.");
+                }
+            }).run();
+
+        // When
+        List<Socket> testClients = BattleshipServer.connectWithPlayers(serverSocket);
+
```
Comment (paulstarbi) [inline src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTest.java]: Nice thread test

**Comment by patkkka** [comment]
Pull request merge conflicts resolved via commad line interface -> merged into v2 and then into master branch manually.

**Pull request closed without merging**